torch/csrc/DeviceAccelerator.cpp

```diff
@@ -13,68 +12,52 @@ void initModule(PyObject* module) {
   });
 
   m.def("_accelerator_deviceCount", []() {
-    const auto device_type = at::getAccelerator(false);
-    if (!device_type.has_value()) {
-      return static_cast<c10::DeviceIndex>(0);
-    }
-    torch::utils::maybe_initialize_device(device_type.value());
-    c10::impl::VirtualGuardImpl impl(device_type.value());
-    return static_cast<c10::DeviceIndex>(impl.deviceCount());
+    auto device_type = at::accelerator::getAccelerator(false);
+    torch::utils::maybe_initialize_device(device_type);
+    return at::accelerator::deviceCount();
   });
 
   m.def("_accelerator_setDeviceIndex", [](c10::DeviceIndex device_index) {
-    const auto device_type = at::getAccelerator(true).value();
+    const auto device_type = at::accelerator::getAccelerator(true).value();
```

Comment:
nit: let's move this after the condition so that invalid index won't even trigger the error of no-accelerator here.

Reply:
Thanks, updated.